### PR TITLE
Indev to Master

### PR DESCRIPTION
**Pull request by HoverEpic**
Changelog
- Update to 1.2.7 (protocol 160)
- Add more blocks
- Add more blocks to translator (some data values always missing)
- Working animals and monsters entities (spawn / despawn /move)
- Block place / break works in creative mode
- Fix yaw / pitch at login
- Fix player permissions (set to member)
- Send language during login
- format a lot of classes to fit the coding style

### CODE REVIEW

File: proxy/src/main/java/org/dragonet/proxy/network/translator/ItemBlockTranslator.java [hunk 1]
```diff
@@ -322,6 +323,11 @@ public static ItemStack translateToPC(Slot slot) {
 
         return item;
     }
+    
+    public static BlockFace translateToPC(int face)
```
Comment (lukeeey) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/ItemBlockTranslator.java]: Omfg stop using your own brace style

File: proxy/src/main/java/org/dragonet/proxy/network/cache/EntityCache.java [hunk 2]
```diff
@@ -67,6 +67,10 @@ public CachedEntity getClientEntity() {
 	}
 
 	public CachedEntity getByRemoteEID(long eid) {
+            if (!mapRemoteToClient.containsKey(eid))
```
Comment (lukeeey) [inline proxy/src/main/java/org/dragonet/proxy/network/cache/EntityCache.java]: Either fix your formatting and brace style or don't commit at all

File: proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlayerPositionRotationPacketTranslator.java [hunk 3]
```diff
@@ -68,6 +75,15 @@ public PCPlayerPositionRotationPacketTranslator() {
 				ret.premiumWorldTemplateId = "";
 				session.sendPacket(ret);
 			}
+                        
+                        ByteArrayDataOutput out = ByteStreams.newDataOutput();
+                        out.writeUTF("DragonProxy");
+                        ClientPluginMessagePacket ClientPluginMessagePacket = new ClientPluginMessagePacket("MC|Brand" , out.toByteArray());
```
Comment (lukeeey) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCPlayerPositionRotationPacketTranslator.java]: Formatting again.

File: proxy/src/main/java/org/dragonet/proxy/utilities/LoginChainDecoder.java [hunk 4]
```diff
@@ -57,6 +58,8 @@ public void decode() {
 
 		// client data
 		clientData = decodeToken(new String(clientDataJWT, StandardCharsets.UTF_8));
+                
+                language = clientData.has("LanguageCode") ? clientData.get("LanguageCode").getAsString() : "en_US";
```
Comment (lukeeey) [inline proxy/src/main/java/org/dragonet/proxy/utilities/LoginChainDecoder.java]: omfg

File: proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCAnimationPacketTranslator.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	// vars
+
+	// constructor
+	public PCAnimationPacketTranslator() {
+
+	}
+
+	// public
+	public PEPacket[] translate(UpstreamSession session, ServerEntityAnimationPacket packet) {
+
```
Comment (lukeeey) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCAnimationPacketTranslator.java]: Are you serious? Don't copy entire classes from other projects.

File: proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCSoundEventPacketTranslator.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+    }
+
+    // public
+    public PEPacket[] translate(UpstreamSession session, ServerPlayBuiltinSoundPacket packet) {
+        LevelSoundEventPacket pk = new LevelSoundEventPacket();
+        pk.position = new Vector3F((float)packet.getX(), (float)packet.getY(), (float)packet.getZ());
+        pk.pitch = (int)packet.getPitch();
+        switch(packet.getSound())
+        {
```
Comment (lukeeey) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/pc/PCSoundEventPacketTranslator.java]: ...

File: proxy/src/main/java/org/dragonet/proxy/network/translator/pe/PEInteractPacketTranslator.java [hunk 7]
```diff
@@ -17,23 +17,35 @@
...
 
-	// public
-	public Packet[] translate(UpstreamSession session, InteractPacket packet) {
-		ClientPlayerInteractEntityPacket pk = new ClientPlayerInteractEntityPacket(
-				(int) (packet.targetRtid & 0xFFFFFFFF), InteractAction.ATTACK);
-		return new Packet[] { pk };
-	}
+    // public
+    public Packet[] translate(UpstreamSession session, InteractPacket packet) {
+        CachedEntity e = session.getEntityCache().getByLocalEID(packet.targetRtid);
```
Comment (lukeeey) [inline proxy/src/main/java/org/dragonet/proxy/network/translator/pe/PEInteractPacketTranslator.java]: Finally someone who knows how to use the TAB button!

**Review by lukeeey — CHANGES_REQUESTED** [review]
Ugh

File: proxy/src/main/java/org/dragonet/proxy/network/cache/EntityCache.java (hunk 2 above)
Comment (HoverEpic) [reply proxy/src/main/java/org/dragonet/proxy/network/cache/EntityCache.java]: file reformated

File: proxy/src/main/java/org/dragonet/proxy/network/translator/pe/PEInteractPacketTranslator.java (hunk 7 above)
Comment (HoverEpic) [reply proxy/src/main/java/org/dragonet/proxy/network/translator/pe/PEInteractPacketTranslator.java]: For explanations : the entire project is coded with `\t`, wich is horrible.
I'm replacing all by four spaces tabs when I hardly work on a class.
I'll do the conversion on all classes i pass now.

**Comment by lukeeey** [comment]
When compiling this branch i get an error:

```
java.lang.ClassNotFoundException: org.apache.maven.cli.MavenCli
        at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
        at org.codehaus.plexus.classworlds.launcher.Launcher.getMainClass(Launcher.java:144)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:266)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at **org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)**
```

**Comment by HoverEpic** [comment]
What version of maven do you use? Can you try the latest?

**Comment by lukeeey** [comment]
Its on my PC and im not at my PC rn so idk

**Review by lukeeey — APPROVED** [review]
Looks alright to me

**Comment by tr7zw** [comment]
I just tried the Indev branch, and as far as I can see it has tons of bugs compared to master?
Item translation is wrong, some Entities get placed at the wrong location, some blocks are rotated worse than in master and the connection desyncs nearly instantly. Does the Project have a Discord/IRC, because I feel like I'm missing something?

**Pull request merged**